packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -658,17 +658,17 @@ const calcIgnorePatternRegExp = (config: Config.ProjectConfig) => {
   return new RegExp(config.transformIgnorePatterns.join('|'));
 };
 
-const calcTransformRegExp = (config: Config.ProjectConfig) => {
-  if (!config.transform.length) {
+const calcTransformRegExp = ({transform = []}: Config.ProjectConfig) => {
```

Comment:
I've done this in lieu of changing every test that does `{ ... } as Config.ProjectConfig` to include `transform: []`.

It technically means that `transform` could be marked as optional, but have left that alone for now as it would just create more work, and probably isn't desired anyway.

Let me know if you're happy with this, or would prefer a different solution :)